src/server/req_stat.c

```diff
@@ -767,41 +795,35 @@ req_stat_sched(struct batch_request *preq)
 	svrattrl	   *pal;
 	struct batch_reply *preply;
 	struct brp_status  *pstat;
-	static char objname[] = "scheduler@";
-
+	int rc =0;
 
 	/* allocate a reply structure and a status sub-structure */
 
 	preply = &preq->rq_reply;
 	preply->brp_choice = BATCH_REPLY_CHOICE_Status;
 	CLEAR_HEAD(preply->brp_un.brp_status);
 
-	pstat = (struct brp_status *)malloc(sizeof(struct brp_status));
-	if (pstat == (struct brp_status *)0) {
-		reply_free(preply);
-		req_reject(PBSE_SYSTEM, 0, preq);
-		return;
+	pbs_sched *psched = NULL;
+	if(strlen(preq->rq_ind.rq_status.rq_id) != 0) {
+		psched = find_scheduler(preq->rq_ind.rq_status.rq_id);
+		if(psched) {
+			rc = status_sched(psched, preq, &preply->brp_un.brp_status);
```

Comment:
when there is a failure in status_sched then we do not use rc to reject the request.